site/_data/lib/find.js

```diff
@@ -74,15 +65,54 @@ const findByUrl = (collection, url, locale = '') => {
   }
 
   const urlToFind = path.join(locale, url);
-  return internalFind(collection, urlToFind);
+  return internalFind(collection, urlCacheKey, 'url', urlToFind);
 };
 
 const findByProjectKey = (collection, projectKey, locale) => {
   if (!projectKey || !locale) {
     return undefined;
   }
   const urlToFind = path.join('/', locale, '/docs/', projectKey, '/');
-  return internalFind(collection, urlToFind);
+  return internalFind(collection, urlCacheKey, 'url', urlToFind);
+};
+
+const findByFilePath = (collection, filePath, locale = '') => {
```

Comment:
Please add a doc here, especially regarding what is considered root for the filePath here (. or /site or ...)?

Reply:
Documented both this and the already existing findByUrl.